lib/sinon/default-behaviors.js

```diff
@@ -181,6 +183,12 @@ module.exports = {
         return fake;
     },
 
+    resolvesThis: function resolvesThis(fake) {
+        fake.resolveThis = true;
+        fake.resolve = false;
+        fake.reject = false;
```

Comment:
IMO we should keep this consistent with other behaviors, such as `rejects`, which also sets many other flags, as you can see [here](https://github.com/sinonjs/sinon/pull/1517/files#diff-f46d81a3e31504be425e201260dab281R165). Even though this is enough to make it work, I think that by keeping the whole API consistent when it comes to flagging fakes is extremely important.